tests/modules/images/fetchpriority/Fetchpriority-Tests.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+class Fetchpriority_Tests extends WP_UnitTestCase {
+
+	public function test_wp_filter_content_tags_filter_adds_fetchpriority_fist_image_only() {
+		$attachment_id = $this->factory->attachment->create_upload_object( TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/leafs.jpg' );
+		$img           = get_image_tag( $attachment_id, '', '', '', 'large' );
+
+		$this->assertStringContainsString( 'fetchpriority="high"', fetchpriority_img_tag_add( $img, 'the_content' ) );
+		$this->assertStringNotContainsString( 'fetchpriority="high"', fetchpriority_img_tag_add( $img, 'not_content' ) );
+
+		$img = str_replace( '<img ', '<img loading="lazy" ', $img );
+		$this->assertStringNotContainsString( 'fetchpriority="high"', fetchpriority_img_tag_add( $img, 'the_content' ) );
+	}
+}
```

Comment:
We should add another test here that actually calls `wp_filter_content_tags()` to test the integration, since your test above only tests direct calls to the newly added function. For the new test, let's pass some demo content with at least two images in it. The lazy-loading shouldn't matter here. We just need to ensure that the first image has the `fetchpriority` attribute while the second one shouldn't, to test that the removal of the filter works.

Also see my comment above around adding one test for the featured image integration.

Reply:
this is in place as `test_fetchpriority_img_tag_add_in_wp_filter_content_tags`